controllers/front/notify.php

```diff
@@ -426,47 +467,57 @@ public function rollbackMerchantOrder()
 
     /**
      * Lock the concurrency to prevent duplicated inputs
-     *
      * @param $orderId
-     * @return bool|void
-     * @throws ConcurrencyException
+     *
+     * @return bool
+     * @throws UnknownException
      */
     protected function blockConcurrency($orderId)
     {
         try {
-            $table = 'pagantis_cart_process';
+            $table = self::PAGANTIS_CART_TABLE;
             if (Db::getInstance()->insert($table, array('id' => (int)$orderId, 'timestamp' => (time()))) === false) {
-                if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+                if ($this->isNotification()) {
                     throw new ConcurrencyException();
+                } else {
+                    $query = sprintf(
+                        "SELECT TIMESTAMPDIFF(SECOND,NOW()-INTERVAL %s SECOND, FROM_UNIXTIME(timestamp)) 
+                                as rest FROM %s WHERE %s",
+                        self::CONCURRENCY_TIMEOUT,
+                        _DB_PREFIX_.$table,
+                        'id='.(int)$orderId
+                    );
+                    $resultSeconds = Db::getInstance()->getValue($query);
+                    $restSeconds = isset($resultSeconds) ? ($resultSeconds) : 0;
+                    $secondsToExpire = ($restSeconds>self::CONCURRENCY_TIMEOUT) ?
+                        self::CONCURRENCY_TIMEOUT : $restSeconds;
+                    if ($secondsToExpire > 0) {
+                        sleep($secondsToExpire + 1);
+                    }
+
+                    $this->getMerchantOrderId();
```

Comment:
Ahora que reviso puede haber un error de concepto a la hora de llamar a las variables. Una cosa es el cartId, que es el que viene en la url, y otra cosa es el orderId una vez se ha guardado la compra en Prestashop.
Mi intención es que aquí si ya está la order guardada en prestashop, se obtenga de nuestra tabla el orderId de prestashop, para que se imprima en el json de la notificación(L.125). Esto hará también modificar la línea 584/595=>$this->module->currentOrder, porque no siempre module estará seteado